pkg/overlay/server_test.go

```diff
@@ -65,3 +67,142 @@ func TestServer(t *testing.T) {
 		}
 	}
 }
+
+func TestNewNodeFiltering(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.New(t, 1, 4, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Check(planet.Shutdown)
+
+	planet.Start(ctx)
+	// we wait a second for all the nodes to complete bootstrapping off the satellite
+	time.Sleep(2 * time.Second)
+
+	satellite := planet.Satellites[0]
+
+	for i, tt := range []struct {
+		name                  string
+		newNodeAuditThreshold int64
+		newNodePercentage     float64
+		requestedNodeAmt      int64
+		expectedResultLength  int
+		reputableNodes        int
+	}{
+		{
+			name:                  "case: fewer than required reputable nodes",
+			requestedNodeAmt:      4,
+			reputableNodes:        3,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: more than required reputable nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        3,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: zero reputable nodes found, only new nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        0,
+			expectedResultLength:  2,
```

Comment:
shouldn't `expectedResultLength` be 0 since `newNodePercentage` is 0 for this case?

Reply:
yeahh since I was planning on coming back to this, I was being sneaky by putting 2 as expectedResultLength but it should've been 0. but that would fail, and I'm not sure if it's the test or the filtering code so I'll check it out.